[ews-build.webkit.org] Formally block PRs when ValidateSquashed fails

JonWBedard opened the pull request:
#### 4dfcc954e0763f80bc6c90f428443f0b7c862e30
<pre>
[ews-build.webkit.org] Formally block PRs when ValidateSquashed fails
<a href="https://bugs.webkit.org/show_bug.cgi?id=242876">https://bugs.webkit.org/show_bug.cgi?id=242876</a>
&lt;rdar://problem/97229223&gt;

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(ValidateSquashed):
(ValidateSquashed.__init__): Set default summary.
(ValidateSquashed.getResultSummary): Use self.summary to set step description.
(ValidateSquashed.evaluateCommand): Annotate the associated PR or bug on failure.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/252597@main">https://commits.webkit.org/252597@main</a>
</pre>

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4847,20 +4849,38 @@ def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
-            return {'step': 'Can only land squashed branches'}
-        if self.results == SUCCESS:
-            return {'step': 'Verified branch is squashed'}
+        if self.results in (FAILURE, SUCCESS):
```
this if isn't required. self.summary is set for all the cases

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4847,20 +4849,38 @@ def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
         if rc != SUCCESS:
+            self.summary = 'Failed to check if commit is squashed'
+            self.setProperty('build_finish_summary', self.summary)
+            self.build.addStepsAfterCurrentStep([RemoveFlagsOnPatch(), RemoveLabelsFromPullRequest()])
```
assuming this is infrastructure failure, we should comment as well asking users to retry. Or we can automatically retry

JonWBedard replied:
I'll add a comment asking for the user to retry.

I've never actually seen us encounter this failure mode, I'm not sure how it's possible.

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4847,20 +4849,38 @@ def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
+            self.summary = 'Verified commit is squashed'
             return SUCCESS
+
+        self.summary = 'Can only land squashed commits'
```
Can word it from user's perspective. e.g.:

`This PR contains multiple commits which are not squashed together. Please squash the commits.`

aj062 reviewed (approved):
r+ with comments

JonWBedard committed:
[ews-build.webkit.org] Formally block PRs when ValidateSquashed fails
https://bugs.webkit.org/show_bug.cgi?id=242876
<rdar://problem/97229223>

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(ValidateSquashed):
(ValidateSquashed.__init__): Set default summary.
(ValidateSquashed.getResultSummary): Use self.summary to set step description.
(ValidateSquashed.evaluateCommand): Annotate the associated PR or bug on failure.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/252597@main
